akka-stream/src/main/scala/akka/stream/impl/Throttle.scala

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import akka.stream.ThrottleMode.{ EnforcingMode, ShapingMode }
+import akka.stream.impl.fusing.GraphStages.SimpleLinearGraphStage
+import akka.stream.stage._
+import akka.stream._
+
+import scala.concurrent.duration.{ FiniteDuration, _ }
+import scala.util.control.NonFatal
+
+/**
+ * INTERNAL API
+ */
+private[stream] trait ThrottleTimeCalculationHelper {
```

Comment:
just for having time conversion logic separate. Still can merge it to AbstractThrottle